core/crypto/src/signature.rs

```diff
@@ -255,10 +284,33 @@ impl TryFrom<&str> for PublicKey {
     }
 }
 
+#[derive(Clone)]
+// This is actually a keypair, because ed25519_dalek api only has keypair.sign
```

Comment:
This comment is not clear. What is `keypair.sign`? Why does it imply that this array is a keypair?

Reply:
I think it is an array that is the concatenation of private and public keys.

Reply:
Yes. this lib has PublicKey, SecretKey and KeyPair. Only KeyPair has `sign` method. If I store secretkey here, i would have to secretkey => get pubkey from secret key => create keypair  with pub/sec key => sign

Reply:
Please then explain the format of this array. How public and private keys are concatenated, which one goes first, what are their lenghts.